docs/Manpage.md

```diff
@@ -2201,6 +2203,9 @@ example, run `export HOMEBREW_NO_INSECURE_REDIRECT=1` rather than just
 - `HOMEBREW_DEBUG`
   <br>If set, always assume `--debug` when running commands.
 
+- `HOMEBREW_SEQUENTIAL_FETCH`
```

Comment:
Do you want to set this or use the flag? Feels like we only need one.

Reply:
> I'm not sure what I think about this, I'm afraid. I don't really like that for this single feature/user we're adding a new SSH flag and a new way of running `brew update`. Honestly it makes me tempted to revert the previous SSH PR if it doesn't meet your use-case.
> 
> I'm also worried that even this more invasive approach will not fix things for you.
> 
> Honestly, I think that using SSH for your private repositories and authenticating in this way is just not a good fit for Homebrew.

I see where you're coming from - adding a new flag for a small use case is a fast road to polluting the update command with god knows what else. Perhaps, then, the environment variable is enough. My midnight thoughts were that this would change the flow enough to be worthy of a flag.

The reason I added the environment variable to the docs too is because I noticed that sometimes other brew commands will call `brew update` too, without the flag of course. Now that I think about it more, that being the case, an environment variable is really the appropriate thing here.

I think the use case is significant enough to support it in Brew. Hardware SSH keys are (in my opinion) the safest way to secure access to a private Git repo. GitHub thought so too and explicitly added support for it last year: https://github.blog/2021-05-10-security-keys-supported-ssh-git-operations/

Would a refactor that only has the environment variable be acceptable @MikeMcQuaid?

Thank you again!

Reply:
> I think the use case is significant enough to support it in Brew. Hardware SSH keys are (in my opinion) the safest way to secure access to a private Git repo. GitHub thought so too and explicitly added support for it last year: https://github.blog/2021-05-10-security-keys-supported-ssh-git-operations/

Honestly, it sounds to me like this user experience is pretty awful (press button once per private repo every time update is run automatically or manually) and very niche (you're the only person I've heard of that wants this) so I'm pretty reluctant to add this unless other @Homebrew/maintainers feel very strongly about it being included.

I've made some reverts in https://github.com/Homebrew/brew/pull/13483 as it sounds like the previous PR isn't useful to you as-is.

---

If I can be convinced by other maintainers this should be included the changes I think are necessary are:
- add back the forced SSH env and perhaps make this env more specifically refer to hardware keys and set the forced SSH env too, so people don't need to set both
- remove the flag here
- ensure that it's tested sufficiently that another follow-up PR isn't needed

I would strongly advise not working on the above until you get a 👍🏻 from me, though, because I want to make sure you don't waste your time ❤️ 

Sorry about all this, I know it must be frustrating but I have to balance your use-case with keeping this code maintainable and supporting millions of Homebrew users.